target_chains/cosmwasm/tools/src/chains-manager/cosmwasm.ts

```diff
@@ -16,48 +21,61 @@ import {
   DeliverTxResponse,
   MsgExecuteContractEncodeObject,
   MsgInstantiateContractEncodeObject,
-  MsgStoreCodeEncodeObject,
-  SigningCosmWasmClient,
   MsgMigrateContractEncodeObject,
+  MsgStoreCodeEncodeObject,
   MsgUpdateAdminEncodeObject,
+  SigningCosmWasmClient,
 } from "@cosmjs/cosmwasm-stargate";
-import { GasPrice, calculateFee } from "@cosmjs/stargate";
+import { GasPrice } from "@cosmjs/stargate";
 import Long from "long";
 import assert from "assert";
 
 export class CosmwasmExecutor implements ChainExecutor {
   constructor(
     private readonly endpoint: string,
-    private readonly mnemonic: string,
-    // chain addresses prefix
-    // example osmo
-    private readonly prefix: string,
+    private readonly signer: OfflineSigner,
     // example - 0.025uosmo
     private readonly gasPrice: string
   ) {}
 
-  private async getAddress(): Promise<string> {
-    const wallet = await DirectSecp256k1HdWallet.fromMnemonic(this.mnemonic, {
-      prefix: this.prefix,
-    });
+  /**
+   * Returns a signer from a mnemonic and prefix to use for the executor
+   * @param mnemonic
+   * @param prefix chain address prefix (example - osmo)
+   */
+  static async getSignerFromMnemonic(mnemonic: string, prefix: string) {
+    const directSecp256k1HdWallet = await DirectSecp256k1HdWallet.fromMnemonic(
+      mnemonic,
+      { prefix }
+    );
+    await directSecp256k1HdWallet.getAccounts();
```

Comment:
why this call is needed? Can you document it? 🤔

Reply:
good catch